support/testing/scenarios/setup.go

```diff
@@ -0,0 +1,150 @@
+	require.NoError(t, err)
+
+	FIL := big.NewInt(1e18)
+
+	initializeActor(ctx, t, vm, &system.State{}, builtin.SystemActorCodeID, builtin.SystemActorAddr, big.Zero())
+
+	addrTreeRoot, accountAddrs := initAddressTree(t, vm, rootVerifier, additionalAccounts)
+	initState := initactor.ConstructState(addrTreeRoot, "scenarios")
+	initState.NextID = abi.ActorID(builtin.FirstNonSingletonActorId + additionalAccounts)
+	initializeActor(ctx, t, vm, initState, builtin.InitActorCodeID, builtin.InitActorAddr, big.Zero())
+
+	rewardState := reward.ConstructState(big.Mul(big.NewInt(100), FIL))
+	initializeActor(ctx, t, vm, rewardState, builtin.RewardActorCodeID, builtin.RewardActorAddr, big.Max(big.NewInt(14e8), FIL))
+
+	cronState := cron.ConstructState(cron.BuiltInEntries())
+	initializeActor(ctx, t, vm, cronState, builtin.CronActorCodeID, builtin.CronActorAddr, big.Zero())
+
+	powerState := power.ConstructState(emptyMapCID, emptyMultimapCID)
+	initializeActor(ctx, t, vm, powerState, builtin.StoragePowerActorCodeID, builtin.StoragePowerActorAddr, big.Zero())
+
+	marketState := market.ConstructState(emptyMapCID, emptyArrayCID, emptyMultimapCID)
+	initializeActor(ctx, t, vm, marketState, builtin.StorageMarketActorCodeID, builtin.StorageMarketActorAddr, big.Zero())
+
+	vrState := verifreg.ConstructState(emptyMapCID, rootVerifier.idAddr)
+	initializeActor(ctx, t, vm, vrState, builtin.VerifiedRegistryActorCodeID, builtin.VerifiedRegistryActorAddr, big.Zero())
+
+	// burnt funds
+	initializeActor(ctx, t, vm, &account.State{Address: builtin.BurntFundsActorAddr}, builtin.AccountActorCodeID, builtin.BurntFundsActorAddr, big.Zero())
+
+	// verified root
+	rootVerifierState := &account.State{Address: rootVerifier.pubAddr}
+	initializeActor(ctx, t, vm, rootVerifierState, builtin.AccountActorCodeID, rootVerifier.idAddr, big.Zero())
+
+	// additional accounts
+	initialAccountBalance := big.Mul(big.NewInt(10_000), FIL)
+	pubAddrs := make([]address.Address, len(accountAddrs))
+	for i, addrPair := range accountAddrs {
+		st := &account.State{Address: addrPair.pubAddr}
+		initializeActor(ctx, t, vm, st, builtin.AccountActorCodeID, addrPair.idAddr, initialAccountBalance)
+		pubAddrs[i] = addrPair.pubAddr
+	}
+
+	_, err = vm.commit()
+	require.NoError(t, err)
+
+	return vm, pubAddrs
+}
+
+func initializeActor(ctx context.Context, t *testing.T, vm *VM, state runtime.CBORMarshaler, code cid.Cid, a address.Address, balance abi.TokenAmount) {
+	stateCID, err := vm.store.Put(ctx, state)
+	require.NoError(t, err)
+	actor := &TestActor{
+		Head:    stateCID,
+		Code:    code,
+		Balance: balance,
+	}
+	err = vm.setActor(ctx, a, actor)
+	require.NoError(t, err)
+}
+
+type addrPair struct {
+	pubAddr address.Address
+	idAddr  address.Address
+}
+
+func initAddressTree(t *testing.T, vm *VM, rootVerifier addrPair, additionalAccounts int) (cid.Cid, []addrPair) {
+	m := adt.MakeEmptyMap(vm.store)
+	err := m.Put(adt.AddrKey(rootVerifier.pubAddr), cborAddrID(t, rootVerifier.idAddr))
```

Comment:
I don't understand the restrictions of the verified registry actor enough to know why this is here, but neither will ~anyone else. Could you please add some more commentary about what and why is happening here?

Reply:
Added. The simple version is that verified registry references a special account actor which needs to be initialized prior to initializing the registry. We now handle this one-off separately from initializing accounts used for testing.